File: scala/conversion/src/org/jetbrains/plugins/scala/conversion/copy/RestoreReferencesDialog.java
package org.jetbrains.plugins.scala.conversion.copy;

import com.intellij.CommonBundle;
import com.intellij.codeInsight.CodeInsightBundle;
import com.intellij.ide.util.FQNameCellRenderer;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import com.intellij.openapi.ui.VerticalFlowLayout;
import com.intellij.psi.PsiClass;
import com.intellij.ui.ScrollPaneFactory;
import com.intellij.ui.components.JBLabel;
import com.intellij.ui.components.JBList;
import com.intellij.util.ui.UIUtil;

import javax.swing.*;
import java.awt.*;
import java.util.Collections;
import java.util.List;

import static com.intellij.util.ui.UIUtil.ComponentStyle.SMALL;
import static com.intellij.util.ui.UIUtil.FontColor.BRIGHTER;

@SuppressWarnings(value = "unchecked")
class RestoreReferencesDialog extends DialogWrapper {
  private final String[] myNamedElements;
  private JList<String> myList;
  private List<String> mySelectedElements = Collections.emptyList();
  private boolean myContainsClassesOnly = true;

  public RestoreReferencesDialog(final Project project, final String[] elements) {
    super(project, true);
    myNamedElements = elements;
    for (Object element : elements) {
      if (!(element instanceof PsiClass)) {
        myContainsClassesOnly = false;
        break;
      }
    }
    if (myContainsClassesOnly) {
      setTitle(CodeInsightBundle.message("dialog.import.on.paste.title"));
    }
    else {
      setTitle(CodeInsightBundle.message("dialog.import.on.paste.title2"));
    }
    init();

    myList.setSelectionInterval(0, myNamedElements.length - 1);
  }

  @Override
  protected void doOKAction() {
    mySelectedElements = myList.getSelectedValuesList();
    super.doOKAction();
  }

  @Override
  protected JComponent createCenterPanel() {
    final JPanel panel = new JPanel(new BorderLayout(UIUtil.DEFAULT_HGAP, UIUtil.DEFAULT_VGAP));
    myList = new JBList<>(myNamedElements);
    myList.setCellRenderer(new FQNameCellRenderer());
    panel.add(ScrollPaneFactory.createScrollPane(myList), BorderLayout.CENTER);

    panel.add(new JBLabel(myContainsClassesOnly ?
                          CodeInsightBundle.message("dialog.paste.on.import.text") :
                          CodeInsightBundle.message("dialog.paste.on.import.text2"), SMALL, BRIGHTER), BorderLayout.NORTH);

    final JPanel buttonPanel = new JPanel(new VerticalFlowLayout());
    final JButton okButton = new JButton(CommonBundle.getOkButtonText());
    getRootPane().setDefaultButton(okButton);
    buttonPanel.add(okButton);
    final JButton cancelButton = new JButton(CommonBundle.getCancelButtonText());
    buttonPanel.add(cancelButton);

    panel.setPreferredSize(new Dimension(500, 400));

    return panel;
  }


  @Override
  protected String getDimensionServiceKey(){
    return "#com.intellij.codeInsight.editorActions.RestoreReferencesDialog";
  }

  public List<String> getSelectedElements(){
    return mySelectedElements;
  }
}

File: scala/conversion/src/org/jetbrains/plugins/scala/conversion/copy/RestoreReferencesDialog.scala
package org.jetbrains.plugins.scala.conversion.copy

import com.intellij.CommonBundle
import com.intellij.java.JavaBundle
import com.intellij.openapi.editor.colors.EditorColorsScheme
import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.{DialogWrapper, VerticalFlowLayout}
import com.intellij.ui.ScrollPaneFactory
import com.intellij.ui.components.{JBLabel, JBList}
import com.intellij.util.ui.UIUtil
import com.intellij.util.ui.UIUtil.ComponentStyle.SMALL
import com.intellij.util.ui.UIUtil.FontColor.BRIGHTER
import org.jetbrains.plugins.scala.lang.refactoring.Associations
import org.jetbrains.plugins.scala.project.ScalaFeatures

import java.awt._
import java.util.Collections
import javax.swing._
import scala.jdk.CollectionConverters.ListHasAsScala

class RestoreReferencesDialog(
  project: Project,
  bindings: Seq[Associations.Binding],
  features: ScalaFeatures,
  colorScheme: EditorColorsScheme
) extends DialogWrapper(project, true) {

  private val importedPathsArray = bindings.toArray

  override protected def getDimensionServiceKey = "#com.intellij.codeInsight.editorActions.RestoreReferencesDialog"

  private var myList: JList[Associations.Binding] = _
  private var mySelectedElements: java.util.List[Associations.Binding] = Collections.emptyList
  def getSelectedElements: Seq[Associations.Binding] = mySelectedElements.asScala.toSeq

  setTitle(JavaBundle.message("dialog.import.on.paste.title2"))

  init()

  myList.setSelectionInterval(0, importedPathsArray.length - 1)

  override protected def doOKAction(): Unit = {
    mySelectedElements = myList.getSelectedValuesList
    super.doOKAction()
  }

  override protected def createCenterPanel: JComponent = {
    val panel = new JPanel(new BorderLayout(UIUtil.DEFAULT_HGAP, UIUtil.DEFAULT_VGAP))
    myList = new JBList(importedPathsArray: _*)
    myList.setCellRenderer(new BindingCellRenderer(features, colorScheme, project))
    panel.add(ScrollPaneFactory.createScrollPane(myList), BorderLayout.CENTER)
    panel.add(new JBLabel(JavaBundle.message("dialog.paste.on.import.text2"), SMALL, BRIGHTER), BorderLayout.NORTH)
    val buttonPanel = new JPanel(new VerticalFlowLayout)
    val okButton = new JButton(CommonBundle.getOkButtonText)
    getRootPane.setDefaultButton(okButton)
    buttonPanel.add(okButton)
    val cancelButton = new JButton(CommonBundle.getCancelButtonText)
    buttonPanel.add(cancelButton)
    panel.setPreferredSize(new Dimension(500, 400))
    panel
  }
}